src/PrestaShopBundle/Controller/Admin/Sell/CustomerService/MerchandiseReturnController.php

```diff
@@ -119,14 +135,106 @@ public function editAction(int $orderReturnId, Request $request): Response
             return $this->redirectToRoute('admin_merchandise_returns_index');
         }
 
-        return $this->render('@PrestaShop/Admin/Sell/CustomerService/OrderReturn/edit.html.twig', [
+        $allowPrintingOrderReturnPdf =
+            $editableOrderReturn->getOrderReturnStateId() === OrderReturnSettings::ORDER_RETURN_STATE_WAITING_FOR_PACKAGE_ID;
+
+        return $this->render('@PrestaShop/Admin/Sell/CustomerService/MerchandiseReturn/edit.html.twig', [
+            'allowPrintingOrderReturnPdf' => $allowPrintingOrderReturnPdf,
+            'editableOrderReturn' => $editableOrderReturn,
             'orderReturnForm' => $form->createView(),
+            'orderReturnsProductsGrid' => $this->presentGrid($gridFactory->getGrid($filters)),
             'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
             'enableSidebar' => true,
             'layoutTitle' => $this->trans('Return Merchandise Authorization (RMA)', 'Admin.Orderscustomers.Feature'),
         ]);
     }
 
+    /**
+     * @AdminSecurity("is_granted('delete', request.get('_legacy_controller'))", redirectRoute="admin_merchandise_returns_index")
+     *
+     * @param Request $request
+     * @param int $orderReturnId
+     * @param int $orderReturnDetailId
+     *
+     * @return RedirectResponse
+     */
+    public function deleteProductAction(Request $request, int $orderReturnId, int $orderReturnDetailId): RedirectResponse
+    {
+        try {
+            $this->getCommandBus()->handle(
+                new DeleteProductFromOrderReturnCommand($orderReturnId, $orderReturnDetailId)
+            );
+
+            $this->addFlash(
+                'success',
+                $this->trans('Successful deletion', 'Admin.Notifications.Success')
+            );
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+        }
+
+        return $this->redirectToRoute(
+            'admin_order_returns_edit',
+            [
+                'orderReturnId' => $orderReturnId,
+            ]
+        );
```

Comment:
just optional a style suggestion, I think its ok to do it in one line when there are no more params. 
```suggestion
        return $this->redirectToRoute('admin_order_returns_edit', ['orderReturnId' => $orderReturnId]);
```

Reply:
same can be applied in other places. But not blocking, has no practical impact anyways